streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -112,20 +115,32 @@ public File baseDir() {
      * @throws StreamsException if the store's change log does not contain the partition
      */
     public void register(StateStore store, boolean loggingEnabled, StateRestoreCallback stateRestoreCallback) {
-        if (store.name().equals(CHECKPOINT_FILE_NAME))
+        Map<String, String> sourceStoreToSourceTopic = null;
+        if (topology != null) {
+            sourceStoreToSourceTopic = topology.sourceStoreToSourceTopic();
+        }
+
+        if (store.name().equals(CHECKPOINT_FILE_NAME)) {
             throw new IllegalArgumentException("Illegal store name: " + CHECKPOINT_FILE_NAME);
+        }
 
-        if (this.stores.containsKey(store.name()))
+        if (this.stores.containsKey(store.name())) {
             throw new IllegalArgumentException("Store " + store.name() + " has already been registered.");
+        }
 
-        if (loggingEnabled)
+        if (loggingEnabled) {
             this.loggingEnabled.add(store.name());
-
+        }
+        
         // check that the underlying change log topic exist or not
         String topic;
-        if (loggingEnabled)
+        if (loggingEnabled) {
             topic = storeChangelogTopic(this.applicationId, store.name());
-        else topic = store.name();
+        } else if (sourceStoreToSourceTopic != null && sourceStoreToSourceTopic.containsKey(store.name())) {
+            topic = sourceStoreToSourceTopic.get(store.name());
+        } else {
+            topic = store.name();
```

Comment:
I think we will not have this case after this patch, at least for now; if yes could we just throw an exception and change it in the future if we ever decide to, for example, allow users to setup their changelog topic names?